src/util/address.rs

```diff
@@ -52,10 +52,79 @@ use serde;
 use blockdata::opcodes;
 use blockdata::script;
 use network::constants::Network;
-use consensus::encode;
 use util::base58;
 use util::key;
 
+/// Address error.
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// Base58 encoding error
+    Base58(base58::Error),
+    /// Bech32 encoding error
+    Bech32(bech32::Error),
+    /// The bech32 payload was empty
+    EmptyBech32Payload,
+    /// Script version must be 0 to 16 inclusive
+    InvalidWitnessVersion(u8),
+    /// The witness program must be between 2 and 40 bytes in length.
+    InvalidWitnessProgramLength(usize),
+    /// A v0 witness program must be either of length 20 or 32.
+    InvalidSegwitV0ProgramLength(usize),
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let desc = ::std::error::Error::description(self);
+        match *self {
+            Error::Base58(ref e) => write!(f, "{}: {}", desc, e),
+            Error::Bech32(ref e) => write!(f, "{}: {}", desc, e),
+            Error::InvalidWitnessVersion(v) => write!(f, "{}: {}", desc, v),
+            Error::InvalidWitnessProgramLength(l) => write!(f, "{}: {}", desc, l),
+            Error::InvalidSegwitV0ProgramLength(l) => write!(f, "{}: {}", desc, l),
+            _ => f.write_str(desc),
+        }
+    }
+}
+
+impl ::std::error::Error for Error {
+    fn cause(&self) -> Option<&::std::error::Error> {
+        match *self {
+            Error::Base58(ref e) => Some(e),
+            Error::Bech32(ref e) => Some(e),
+            _ => None,
+        }
+    }
+
+    fn description(&self) -> &str {
```

Comment:
I'd prefer you just return a standard "`Error::description` is deprecated" string and hardcode these values into the `fmt::Display` impl

Reply:
Can I just drop it in that case? It's a provided method.

Reply:
Doesn't seem to be optional in v1.22. I added deprecation mention.